### PR TITLE
Remove `prefect-agent` as a possible work pool type

### PR DESCRIPTION
**Pull request by desertaxle**
Removes `prefect-agent` as a selectable work pool type. `PrefectAgent` has been removed for the 3.0 release, so this is no longer needed. The collection registry will still contain data for the `prefect-agent` work pool type until `PrefectAgent` is removed from the 2.x line.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: tests/server/api/test_collections.py [hunk 1]
```diff
@@ -57,7 +66,7 @@ def mock_get_view(
             return_value=Response(200, json=mock_block_response)
         )
         respx_mock.get(self.collection_view_url("worker")).mock(
-            return_value=Response(404, json=mock_collection_response)
+            return_value=Response(200, json=mock_worker_response)
```
Comment (serinamarie) [inline tests/server/api/test_collections.py:69]: why is this a `200` now?

**Review by serinamarie — APPROVED** [review]
LGTM!

File: tests/server/api/test_collections.py (hunk 1 above)
Comment (desertaxle) [reply tests/server/api/test_collections.py:69]: That mock wasn't being used, so I repurposed it.

Comment (serinamarie) [reply tests/server/api/test_collections.py:69]: gotcha

**Pull request merged**